### ISSUE TITLE
write methods to return bytes object when filepath is None

### ISSUE THREAD

**Issue by d-chambers** [obspy/obspy#1796]
A suggestion for an API improvement:

Currently to to write a file to a bytes object (eg for sending over the network) one must do something like this:
```python
import io

import obspy

st = obspy.read()

with io.BytesIO() as buf:
    st.write(buf, 'mseed')
    buf.seek(0, 0)
    bytes_obj = buf.read()
```

This seems rather verbose and potentially confusing to those not familiar with buffers. Maybe we could automatically return the bytes representation if the file argument is None, something like this:

```python
bytes_obj = st.write(None, 'mseed')
```

Xarray does the same thing with its [to_netcdf](http://xarray.pydata.org/en/stable/generated/xarray.Dataset.to_netcdf.html) method. 

I am happy to work on this but thought I would bring it up before putting any time into it.

**Comment by krischer** [comment]
Good idea! Feel free to give it a shot. Internally it would still have to first write to a `BytesIO` object.

It will also likely not yet work for all file formats as its a bit like the wild west in regards to what each file format can write to but we plan on unifying this once we manage to get the next ObsPy release out. So it should then work for all formats.
